src/main/scala/typechecker/TypeCheck.scala

```diff
@@ -131,7 +131,7 @@ object TypeChecker {
     case EArrLiteral(_) => throw NotInBinder(expr.pos, "Array Literal")
     case ECast(e, castType) => {
       val (typ, nEnv) = checkE(e)
-      if (safeCast(typ, castType) == false) {
+      if (!safeCast(typ, castType)) {
```

Comment:
I've always preferred explicit `== false` instead of `!` which is easy to miss. Also, change not needed in this commit.